temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowOutboundCallsInterceptor.java

```diff
@@ -30,13 +30,34 @@
 import io.temporal.workflow.Workflow;
 import io.temporal.workflow.WorkflowInfo;
 import io.temporal.workflow.unsafe.WorkflowUnsafe;
+import java.util.concurrent.Executor;
 
 public class OpenTracingWorkflowOutboundCallsInterceptor
     extends WorkflowOutboundCallsInterceptorBase {
   private final SpanFactory spanFactory;
   private final Tracer tracer;
   private final ContextAccessor contextAccessor;
 
+  private class SpanTransferringExecutor implements Executor {
+    private final Executor passthrough;
+    private final Span activeSpan;
+
+    public SpanTransferringExecutor(Executor passthrough) {
+      this.passthrough = passthrough;
+      activeSpan = tracer.scopeManager().activeSpan();
+    }
+
+    public void execute(Runnable r) {
+      if (activeSpan != null) {
+        try (Scope ignored = tracer.scopeManager().activate(activeSpan)) {
```

Comment:
For consistency with OTEL and backwards compatibility, If we have an active span here we should not set the `activeSpan`

Reply:
This might be a poor-nomenclature thing than a problem, unless I am misunderstanding you.

In the above, "activeSpan" isnt the current threads activeSpan, but that which was captured during callback/promise creation.  Perhaps I should rename it to something else, like "parentSpan"?  Or are you referring to a different problem?

Reply:
I changed this to "capturedSpan" to be more explicit.  Let me know if you still have concerns here.

Reply:
Yes so i am saying if the thread already has an active span, don't set a new one

Reply:
Ah, I get you now.  I will add that check